agent/main_common.go

```diff
@@ -217,11 +217,16 @@ func debugCheckResults(cfg *config.AgentConfig, check string) error {
 
 func printResults(cfg *config.AgentConfig, ch checks.Check) error {
 	// Run the check once to prime the cache.
-	_, err := ch.Run(cfg, 0)
-	if err != nil {
+	if _, err := ch.Run(cfg, 0); err != nil {
 		return fmt.Errorf("collection error: %s", err)
 	}
-	time.Sleep(1 * time.Second)
+
+	if ch.Name() == checks.Connections.Name() {
+		fmt.Printf("Waiting 5 seconds to allow for active connections to transmit data\n")
```

Comment:
should we have such a wait in the main check as well? basically a warming period where it returns nothing for the first iterations of the check?

Reply:
I don't think it's needed while the agent is actually running since the BPF program will be loaded the entire time.